identity-iota/src/did/doc/iota_verification_method.rs

```diff
@@ -74,22 +75,22 @@ impl IotaVerificationMethod {
     let key: &[u8] = keypair.public().as_ref();
     let did: IotaDID = IotaDID::new_with_network(key, network)?;
 
-    Self::from_did(did, keypair, fragment)
+    Self::from_did(did, keypair.type_(), keypair.public(), fragment)
   }
 
   /// Creates a new [`IotaVerificationMethod`] from the given `did` and `keypair`.
-  pub fn from_did(did: IotaDID, keypair: &KeyPair, fragment: &str) -> Result<Self> {
+  pub fn from_did(did: IotaDID, key_type: KeyType, public_key: &PublicKey, fragment: &str) -> Result<Self> {
```

Comment:
I think you're right about the reason being the Wasm bindings. So in `WasmVerificationMethod` I'll change it to:

```rust
pub fn from_did(did: &WasmDID, key_type: KeyType, public_key: String, fragment: String) -> ...
```

Regarding the default, I don't really like the `from_keypair` method, but doesn't it steer users in the right direction to produce a DID where the initial document is signed by the same key that produced the DID tag?

Reply:
Users typically create documents through the `IotaDocument::new` constructor which derives the `DID` automatically. Users only seem to call `IotaVerificationMethod::new` when trying to add new methods to the document afterwards, which inevitably end up with the incorrect (derived) DID in the `id` field.

Reply:
```
pub fn from_did(did: &WasmDID, key_type: KeyType, public_key: String, fragment: String) -> ...
```

This doesn't solve the problem of accidentally publishing one's `private_key` though which the Rust `PublicKey` type helps prevent. My suggestion would be to expose similar types in Wasm for the `KeyPair` instead of returning strings (and move/rename the string functions).

Reply:
Also, what I don't like about my change here, is that it's possible to provide a different `KeyType` than what the public key represents. We could either store the `KeyType` additionally into the `PublicKey` and `PrivateKey`, or preferably, create a thin wrapper `TypedPublicKey` and `TypedPrivateKey`, that would be the type of the parameter here.

That's probably overdoing... we can also note in the docs that the caller is responsible for providing the right types (it's the low-level API after all).

Reply:
> This doesn't solve the problem of accidentally publishing one's private_key in Wasm

I missed that point in your original post. That is definitely worth changing!

That seems a bit out of scope for this PR, so I'm thinking of reverting the Wasm changes for now and address this properly in a future PR, so the state refactor PR can land sooner.